app/services/printer-service.js

```diff
@@ -11,12 +12,38 @@ export default Ember.Service.extend({
       .sortBy("name")
       .map(printer => printer.getProperties("name", "id"));
   },
+
   updateUserDefaultPrinter(printerId) {
-    new AjaxPromise(
-      `/users/${this.get("session.currentUser.id")}`,
-      "PUT",
-      this.get("session.authToken"),
-      { user: { printer_id: printerId } }
-    ).then(data => this.get("store").pushPayload(data));
+    const defaultPrinterUser = this.__getStockPrintersUsers().get(
```

Comment:
The default printer record might not exist if it's a new user, so this would break.
Maybe it's better to do a "create if missing" behaviour ?

Reply:
Sure.. i'll add it. Thanks

Reply:
We are assigning the first printer  as the default printer if there isn't any for new user.